packages/cli/src/commonlib/codeFlowLogin.ts

```diff
@@ -298,6 +319,57 @@ export class CodeFlowLogin {
     }
   }
 
+  async getTokenByScopes(scopes: Array<string>, refresh = true): Promise<Result<string, FxError>> {
+    try {
+      if (!this.account) {
+        await this.reloadCache();
+      }
+      if (!this.account) {
+        const accessToken = await this.login();
+        return ok(accessToken);
+      } else {
+        return this.pca!.acquireTokenSilent({
+          account: this.account,
+          scopes: scopes,
+          forceRefresh: false,
+        })
+          .then((response) => {
```

Comment:
Is it better to unify promise code with `async`/`await` to make to code more readable? I think it is a bit hard to read when mixing `async`/`await` and `then`/`catch` together.

Reply:
Fixed.